lib/book.py

```diff
@@ -125,117 +112,34 @@ def formats_not_failing(self):
             if info['status'] is not self.FAIL:
                 yield info
 
-    # get book's title, title sort, author, author sort, and asin if it exists
+    # get book's title, title sort, author and author sort if it exists
     def _get_basic_information(self):
         self._title = self._db.field_for('title', self._book_id)
-        self._title_sort = self._db.field_for('sort', self._book_id)
 
         self._author = self._db.field_for('authors', self._book_id)
         if len(self._author) > 0:
             self._author = ' & '.join(self._author)
-        self._author_sort = self._db.field_for('author_sort', self._book_id)
-        if self._title is 'Unknown' or self._title_sort is 'Unknown' or not self._author or not self._author_sort:
+        if self._title is 'Unknown' or not self._author:
             self._status = self.FAIL
             self._status_message = self.FAILED_BASIC_INFORMATION_MISSING
             return
 
-        self._asin = self._book_settings.prefs['asin'] if self._book_settings.prefs['asin'] != '' else None
-        if not self._asin:
-            identifiers = self._db.field_for('identifiers', self._book_id)
-            self._asin = self._db.field_for('identifiers', self._book_id)['mobi-asin'].decode('ascii') if 'mobi-asin' in identifiers.keys() else None
-            if self._asin:
-                self._book_settings.prefs['asin'] = self._asin
-
-        self._shelfari_url = self._book_settings.prefs['shelfari_url'] if self._book_settings.prefs['shelfari_url'] != '' else None
+        self._author_list = list(self._author) if isinstance(self._author, tuple) else self._author
```

Comment:
Not sure if this makes sense a/o works, it might be fine but just confusing me!  I'll comment what I find odd, and you can hopefully say if it does what is intended :)  Perhaps this is just a need for commenting what the DB returns and what we want out..

self._author is set by a DB lookup.  If we have a single author does it return a string or a tuple/list/whatever containing a single string?  I assume in the multi-author case it is a tuple/list/whatever containing the multiple strings.

I assume it has to be returning a tuple-ish item always - otherwise the " & ".join() won't work well for single authors (we'd be getting 'f & o & o & b & a & r' type nonsense).

So depending on what db.field_for("author") returns we'll have the following, I think:
- returns empty tuple -> self._author is empty tuple, but we'll bail out already
- returns empty string -> self._author is empty string, but we'll bail out already
- returns single author as string -> self._author is nonsense, and self._author_list will be the same
- returns a tuple containing author strings -> self._author is "singleauthor", "author1 & author2", etc. and self_author_list is the same (as it won't be a tuple).

I suspect we'll always get a tuple from the DB lookup, in which case we'll bail early if it is zero length (even possible?) and otherwise self._author will be the "author1 & author2" but self._author_list will be the same.  I imagine you wanted self._author_list to be a list - though IMO it should remain a tuple as we shouldn't be modifying it (the whole difference between list and tuple).

Ah - I now see where this is used - in xray_creator.py where we do "if author not in b.author_list".  author_list is strangely not a list but an "author1 & author2" type string, but due to the magic of python this will work, albeit with corner cases you might not expect..

TL;DR: I think DB returns a tuple,  self._author_list should be set to that same tuple (possibly renaming to self._authors for clarity), self._author should be set afterwards to "soloauthor" or "author1 & author2" as you are currently doing (possibly renaming a self._author_string or similar for clarity).

Phew!